ngraph/core/reference/include/ngraph/runtime/reference/ctc_loss.hpp

```diff
@@ -95,82 +95,100 @@ namespace ngraph
 
                     const size_t BTC = b * TC;
 
-                    std::vector<T> kExp(actualLogitLen, 0);
+                    std::vector<std::unordered_map<size_t, T>> probabilities(actualLogitLen);
+                    T prob = 0.f, kExp = 0.f;
                     for (size_t t = 0; t < actualLogitLen; t++)
                     {
-                        size_t btcT = BTC + classesNum * t;
+                        kExp = 0.f;
+                        const size_t btcT = BTC + classesNum * t;
                         for (size_t c = 0; c < classesNum; c++)
                         {
-                            kExp[t] += std::exp(logits[btcT + c]);
+                            kExp += std::exp(logits[btcT + c]);
                         }
+                        for (size_t s = 0; s < decodedTargetLen; s++)
+                        {
+                            prob = std::log(std::exp(logits[btcT + targetD[s]]) / kExp);
```

Comment:
1. Don't you use log softmax formula that is more numerically stable than straightforward formula? 
2. Also, I would rather rename prob to log_prob and below because this is not pure probability.

Reply:
Fixed